### ISSUE TITLE
Regression: pushing a new branch should not trigger checking past commits

### ISSUE THREAD

**Issue by boudekerk** [gnustavo/Git-Hooks#31]
With recent versions we're seeing the is the same behaviour as reported earlier in https://github.com/gnustavo/Git-Hooks/issues/24. This was fixed in 1.12.2, but appears again as of 1.13.1. As a workaround we've reverted back to 1.13.0. The reproduction path from the issue above can be used.

**Comment by boudekerk** [comment]
Updated reproduction path:
```
HOOK=/data/git/bin/githooks.pl
CWD=$(pwd)
git init git-hooks-issue-31-remote
cd git-hooks-issue-31-remote
git checkout -b new-branch
cd ..
git clone file://${CWD}/git-hooks-issue-31-remote git-hooks-issue-31-local
cd git-hooks-issue-31-local
echo ${RANDOM} > README; git add README; git commit -m"Testing" README
git push
echo ${RANDOM} > README; git commit -m"Testing" README
git push
git checkout -b test-branch
ln -sf ${HOOK} ${CWD}/git-hooks-issue-31-remote/.git/hooks/pre-receive
git push -u origin test-branch
```

**Comment by gnustavo** [comment]
Hi, @boudekerk . You're right. Thanks for the report.

I think I fixed it in 7e28924 but I'd like to ask you to validade my change before I can make a new release. Can you give it a try?

**Comment by boudekerk** [comment]
Hi @gnustavo,

your fix indeed makes things work for us as expected. We don't have any post-(receive|update) hooks, so I didn't test whether those also still work.

Kind regards,
Bartosz

**Comment by boudekerk** [comment]
Hi @gnustavo,

do you think you'll be able to create another release containing this fix?

Kind regards,
Bartosz

**Comment by gnustavo** [comment]
Hi @boudekerk ,

I'm sorry for the delay. I just released v1.15.0 which should appear shortly on CPAN.
